plugins/nf-google/src/main/nextflow/cloud/google/batch/GoogleBatchMachineTypeSelector.groovy

```diff
@@ -81,10 +81,10 @@ class GoogleBatchCloudinfoMachineSelector {
         int memPerVm
     }
 
-    static GoogleBatchCloudinfoMachineSelector INSTANCE() {
+    static GoogleBatchMachineTypeSelector INSTANCE() {
         if (instance != null)
             return instance
-        return instance = new GoogleBatchCloudinfoMachineSelector()
+        return instance = new GoogleBatchMachineTypeSelector()
     }
```

Comment:
Even if not harmful in this case, a race condition may arise here. Likely just declaring a static `INSTANCE` field is a better choice

Reply:
done